### PR TITLE
Updates user services to Add keycloak authentication

### LINKED ISSUE #34: Adding Key-Cloak Authentication in User Services
**Describe the bug**
- Adding Key-Cloak Authentication to User API.
- Updating JPA Repositories to add additional required queries.
- Testing existing features

**Expected behavior**
- Once enabled, User Services should be able to validate authentication of new users which are added into the DB.

**Additional Context**
- Need complete test coverage for Repositories and Key-Cloak to make sure everything is working as expected

### PR DESCRIPTION
**Pull request by picksitquick**
Fixes #34 
- added key-cloak auth in User Assembler
- updated test properties
- added additional methods in repositories as per the requirements.

Coverage:
- will be creating certain test methods to test out the repositories and assemblers.
- will verify the behaviour of the project before we move further

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -74,26 +81,26 @@
...
-        <groupId>org.jboss.resteasy</groupId>
-        <artifactId>resteasy-client</artifactId>
-        <version>3.1.3.Final</version>
-    </dependency>
-        <dependency>
-            <groupId>org.mapstruct</groupId>
-            <artifactId>mapstruct</artifactId>
-            <version>1.5.3.Final</version>
-        </dependency>
+<!--        <dependency>-->
```
Comment (aprajshekhar) [inline pom.xml:94]: These dependencies are required for Keycloak.

File: pom.xml [hunk 2]
```diff
@@ -193,32 +200,32 @@
...
-                        </path>
-                        <path>
-                            <groupId>org.projectlombok</groupId>
-                            <artifactId>lombok-mapstruct-binding</artifactId>
-                            <version>0.2.0</version>
-                        </path>
-                    </annotationProcessorPaths>
-                </configuration>
-            </plugin>
+<!--            <plugin>-->
```
Comment (aprajshekhar) [inline pom.xml:203]: Mapper for keycloak related models needs mapstruct.

**Review by aprajshekhar — CHANGES_REQUESTED** [review]
Great job. Just two comments regarding dependencies for Keycloak in pom.xml

File: pom.xml (hunk 2 above)
Comment (picksitquick) [reply pom.xml]: Update made and this can now be merged

**Review by aprajshekhar — APPROVED** [review]
LGTM

**Pull request merged**